### PR TITLE
[RFR] Fix alwaysOn filters defaultValue being ignored

### LINKED ISSUE #1032: defaultValue in Filters is ignored when alwaysOn
Hi again.. is it only me who's not able to set anymore the defaultValue property in any *Input inside a List Filter ..?
For example the default value "test" does not appear:
`const ListFilters = (props) => (
    <Filter {...props} >
      <TextInput label="test" source="test" alwaysOn defaultValue="test" />                             
    </Filter>
);`
Thanks..

### PR DESCRIPTION
**Pull request by djhi**
Fixes #1032 

- [x] The fix
- [x] Tests

### CODE REVIEW

File: src/mui/list/FilterForm.js [hunk 1]
```diff
@@ -113,6 +114,24 @@ FilterForm.defaultProps = {
 
 const enhance = compose(
     translate,
+    withProps(({ initialValues, filters }) => ({
```
Comment (fzaninotto) [inline src/mui/list/FilterForm.js]: I don't understand this one, could you explain the rationale?

Comment (djhi) [reply src/mui/list/FilterForm.js]: Merge `initialialValues` passed by the `List` component (which may have read them from the querystring) with the defaultValues of all `alwaysOn` filters which have a `defaultValue` specified.

Comment (djhi) [reply src/mui/list/FilterForm.js]: `initialialValues` should probably be merged at the end.. back to WIP :)

Comment (djhi) [reply src/mui/list/FilterForm.js]: I confirm that this would have been a prb. Fixed it

**Pull request merged**